### PR TITLE
19 create function data download

### PR DESCRIPTION
**Pull request by soriadelva**
The original idea was to create a function data_download with the same functionality + some extras as the code used in the climate_match script. However, as the input could be either a (vector of) taxonkey(s), a zipfile, or a downloadkey, and these should better not be combined (i.e., it may be confusing to provide both taxonkeys and a zipfile, especially since the zipfile would allow to upload data that may be different from the usual GBIF download as long as the necessary columns are present), I decided to create three seperate functions, one for each type of data download/import. As such, it will also be very clear what the necessary input of each function is (i.e., either taxonkeys, a zipfile, or a downloadkey).

### CODE REVIEW

**Comment by soriadelva** [comment]
It's weird that the checks fail, Devtools::check() did nog signal any errors 🤔

File: R/get_data.R [hunk 1]
```diff
@@ -0,0 +1,657 @@
...
+#'               coord_unc = 100,
+#'               basis_of_record = "HUMAN_OBSERVATION",
+#'               identification_verification_status= c("Probable", "valid","confident","")
+#'               )
+#'
+#' #provide a taxon key and a path to a shapefile
+#' taxon_key<-2427091
+#' url <- "https://zenodo.org/records/3386224"
+#' zipfile <- tempfile(fileext = ".zip")
+#' utils::download.file(url, zipfile, mode = "wb")
```
Comment (SanderDevisscher) [inline R/get_data.R:197]: I get the following error when downloading the shape:
```
Warning message:
In utils::unzip(zipfile, exdir = tempdir()) :
  error 1 in extracting from zip file
```

probably its a missing package I'll investigate.

**Review by SanderDevisscher — CHANGES_REQUESTED** [review]

File: R/get_data.R (hunk 1 above)
Comment (SanderDevisscher) [reply R/get_data.R:197]: use this url: url <- "https://zenodo.org/api/records/3386224/files-archive"

File: tests/testthat/test-get_data.R
Comment (SanderDevisscher) [inline tests/testthat/test-get_data.R:1]: this file is empty ?

File: R/get_data.R [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+#'
+#' @returns An sf data frame holding occurrence records that are ready to be used for climate casting.
+#'
+#' @export
+#'
+#' @examples
+#'  \dontrun{
+#' #provide GBIF taxon_key(s)
+#' taxon_key <- c(2865504, 5274858)
+#' get_gbif_data(taxon_key)
```
Comment (SanderDevisscher) [inline R/get_data.R:26]: @soriadelva any reason you don't store the result of these steps in the environment ? ea add `df <-` ?

**Review by SanderDevisscher — CHANGES_REQUESTED** [review]
Just some small changes and I can approve

File: tests/testthat/test-get_data.R
Comment (soriadelva) [reply tests/testthat/test-get_data.R:1]: Will make the tests in the future, but created the file already

**Review by SanderDevisscher — APPROVED** [review]
good work

**Pull request merged**